xbmc/cores/AudioEngine/Sinks/windows/AESinkFactoryWin32.cpp

```diff
@@ -85,33 +85,66 @@ std::vector<RendererDetail> CAESinkFactoryWin::GetRendererDetails()
     PropVariantInit(&varName);
 
     hr = pEnumDevices->Item(i, pDevice.GetAddressOf());
-    EXIT_ON_FAILURE(hr, "Retrieval of WASAPI endpoint failed.")
+    EXIT_ON_FAILURE(hr, "Retrieval of endpoint failed.")
 
     hr = pDevice->OpenPropertyStore(STGM_READ, pProperty.ReleaseAndGetAddressOf());
-    EXIT_ON_FAILURE(hr, "Retrieval of WASAPI endpoint properties failed.")
+    EXIT_ON_FAILURE(hr, "Retrieval of endpoint properties failed.")
 
     hr = pProperty->GetValue(PKEY_Device_FriendlyName, &varName);
-    EXIT_ON_FAILURE(hr, "Retrieval of WASAPI endpoint device name failed.")
+    EXIT_ON_FAILURE(hr, "Retrieval of endpoint device name failed.")
 
     details.strDescription = KODI::PLATFORM::WINDOWS::FromW(varName.pwszVal);
     PropVariantClear(&varName);
 
     hr = pProperty->GetValue(PKEY_AudioEndpoint_GUID, &varName);
-    EXIT_ON_FAILURE(hr, "Retrieval of WASAPI endpoint GUID failed.")
+    EXIT_ON_FAILURE(hr, "Retrieval of endpoint GUID failed.")
 
     details.strDeviceId = KODI::PLATFORM::WINDOWS::FromW(varName.pwszVal);
     PropVariantClear(&varName);
 
     hr = pProperty->GetValue(PKEY_AudioEndpoint_FormFactor, &varName);
-    EXIT_ON_FAILURE(hr, "Retrieval of WASAPI endpoint form factor failed.")
+    EXIT_ON_FAILURE(hr, "Retrieval of endpoint form factor failed.")
 
     details.strWinDevType = winEndpoints[(EndpointFormFactor)varName.uiVal].winEndpointType;
     details.eDeviceType = winEndpoints[(EndpointFormFactor)varName.uiVal].aeDeviceType;
+    PropVariantClear(&varName);
 
+    /* In shared mode Windows tells us what format the audio must be in. */
+
+    /* Beginning of suspicous code
+     *
+     * Is this seemingly unnecessary activation actually needed in some cases to initialize
+     * PKEY_AudioEngine_DeviceFormat? Or was it forgotten when GetMixFormat() was commented?
+     * Found a blog that indicates that GetMixFormat() can force the population. Maybe only for
+     * older Windows versions?
+     * Doesn't seem needed for Windows 8.1 and above. No mention in MS documentation.
+     * This code came from AESinkDirectSound which was created with this as is.
+     * No prior history found, the DS code prior to 2012 didn't use the MMDevice API.
+     * Kept to avoid breaking something by accident.
+     */
+    ComPtr<IAudioClient> pClient;
+    hr = pDevice->Activate(IID_IAudioClient, CLSCTX_ALL, nullptr,
+                           reinterpret_cast<void**>(pClient.GetAddressOf()));
```

Comment:
If not necessary, why not remove?  This is only master / pre Alpha

Reply:
ok, we'll see.

Reply:
Now I'm almost sure is a leftover of GetMixFormat --> Activate was used to obtain pClient and now pClient is "dead code".

Reply:
Agreed, but it was such a long time ago.., I found a blog saying that GetMixFormat helps populate the property, which put a doubt in my mind. We'll see.